Apps/Common/src/Api/IKeycloakAdminApi.cs

```diff
@@ -46,6 +46,63 @@ public interface IKeycloakAdminApi
         [Get("/roles/{role}/users?first={first}&max={max}")]
         Task<List<UserRepresentation>> GetUsersAsync(string role, int first, int max, [Authorize] string token);
 
+        /// <summary>
+        /// Look up User accounts by username, first or last name, or email.
+        /// </summary>
+        /// <param name="queryString">The text string to search by.</param>
+        /// <param name="first">The first record to return.</param>
+        /// <param name="max">The maximum results to return.</param>
+        /// <param name="token">The bearer token to authorize the call.</param>
+        /// <returns>A List of UserRepresentation objects.</returns>
+        [Get("/users?briefRepresentation=true&first={first}&max={max}&search={queryString}")]
+        Task<List<UserRepresentation>> GetUsersQueryAsync(string queryString, int first, int max, [Authorize] string token);
+
+        /// <summary>
+        /// Adds a user.
+        /// </summary>
+        /// <param name="userRepresentation">The UserRepresentation of new user.</param>
+        /// <param name="token">The bearer token to authorize the call.</param>
+        /// <returns>Returns true when user created.</returns>
+        [Post("/users")]
+        Task AddUserAsync([Body] UserRepresentation userRepresentation, [Authorize] string token);
+
+        /// <summary>
+        /// Get realm roles for user.
+        /// </summary>
+        /// <param name="id">The user's id.</param>
+        /// <param name="token">The bearer token to authorize the call.</param>
+        /// <returns>A List of RoleRepresentation objects.</returns>
+        [Get("/users/{id}/role-mappings/realm")]
+        Task<List<RoleRepresentation>> GetUserRolesAsync(string id, [Authorize] string token);
+
+        /// <summary>
+        /// Adds realm roles to user.
+        /// </summary>
+        /// <param name="id">The user's id.</param>
+        /// <param name="roles">List of RoleRepresentations to add to user.</param>
+        /// <param name="token">The bearer token to authorize the call.</param>
+        /// <returns>Returns true when roles added.</returns>
+        [Post("/users/{id}/role-mappings/realm")]
+        Task AddUserRolesAsync(string id, [Body] IEnumerable<RoleRepresentation> roles, [Authorize] string token);
+
+        /// <summary>
+        /// Adds realm roles to user.
+        /// </summary>
+        /// <param name="id">The user's id.</param>
+        /// <param name="roles">List of RoleRepresentations to delete from the user.</param>
+        /// <param name="token">The bearer token to authorize the call.</param>
+        /// <returns>Returns true when roles deleted.</returns>
+        [Delete("/users/{id}/role-mappings/realm")]
+        Task DeleteUserRolesAsync(string id, [Body] IEnumerable<RoleRepresentation> roles, [Authorize] string token);
```

Comment:
The `id` parameters in these declarations can be changed from `string` to `Guid` to save a lot of `ToString()` calls in the service.